controllers/employeesController.js

```diff
@@ -0,0 +1,142 @@
+const { type } = require('express/lib/response');
+const { check } = require('prettier');
+const db = require('../models');
+
+function getAttributesFromRequest(req){
+	return {
+		name: req.body.name,
+		lastName: req.body.lastName,
+		position: req.body.position
+	}
+}
+function validateEmployee(req, res) {
+	if (attributesInRequest(req, res) && attributesInRequestDefined(req, res) && checkTypeOfAttributes(req, res)) {
+		return true;
+	}
+	return false;
+}
+function attributesInRequest(req, res) {
+	if (emptyObject(req.body)) {
+		res.status(400).send({
+			message: 'No attributes in body',
+		});
+		return false;
+	}
+	return true;
+}
+
+function attributesInRequestDefined(req, res) {
+	let {name: name, lastName: lastName, position: position} = getAttributesFromRequest(req);
+
+	if (name == undefined || lastName == undefined || position == undefined) {
+		res.status(400).send({
+			message: 'No name / last name / position filled',
+		});
+		return false;
+	}
+	return true;
+}
+
+function checkTypeOfAttributes(req, res) {
+	let {name: name, lastName: lastName, position: position} = getAttributesFromRequest(req);
+
+	if (typeof name !== 'string' || typeof lastName !== 'string' || typeof position !== 'string') {
+		res.status(400).send({
+			message: 'Field content cant be a number, they must be characters',
+		});
+		return false;
+	}
+	return true
+}
+
+async function newEmployee(req, res) {
+	if (!validateEmployee(req, res)) {
+		return;
+	}
+
+	let {name: name, lastName: lastName, position: position} = getAttributesFromRequest(req);
+
+	try {
+		await db.empleados.create({
+			nombre: name,
+			apellido: lastName,
+			cargo: position,
+		});
+	} catch (err) {
+		if (err.parent.code == 'ER_NO_DEFAULT_FOR_FIELD') {
+			res.status(400).send({
+				message: 'Not all fields have been filled.',
+				errorMessage: err.parent.sqlMessage,
+			});
+		}
+		res.status(400).send({
```

Comment:
This error could be defined as `500`, because it is a server error (database connection problems for example). 
**Server error codes** indicates that the request was accepted, but that an error on the server prevented the fulfillment of the request.

Reply:
Resolved in 83d7de4ed11bca24c689d1fe577bc38c8e589539